subprojects/docs/src/snippets/maven-publish/conditional-publishing/groovy/build.gradle

```diff
@@ -38,17 +38,23 @@ publishing {
 
 // tag::task-config[]
 tasks.withType(PublishToMavenRepository) {
-    onlyIf("publishing binary to the external repository, or binary and sources to the internal one") {
+    def check = providers.provider {
```

Comment:
💄 Perhaps `spec` since that's the term used in the `onlyIf` documentation?

It would be nice to have an explanation comment as well:
```
    // Capture spec result in the configuration cache
    def spec = ...
```

Reply:
I tend to avoid using variable/parameter names that describe the type, as that is what type declarations are for, and prefer to name after role instead.

I will go with `checkSpec` as a middle ground - stop me if that bothers you.

Reply:
The docs also use the term `predicate`.

Reply:
And by the way, `spec` was not about the type. `spec` is also common noun used in the docs.

Reply:
Cool, went with `predicate`, as a more common name